src/students/students.service.ts

```diff
@@ -26,4 +31,121 @@ export class StudentsService {
   getEmailsFromAddedStudents(addedStudents: Student[]): string[] {
     return addedStudents.map(student => student.user.email);
   }
+
+  // eslint-disable-next-line max-lines-per-function
+  public async getAllStudents(pageOptionsDto: PageOptionsDto): Promise<PageDto<StudentsResponse>> {
+    const queryBuilder = this.getAllStudentsQuery(pageOptionsDto);
+    const itemCount = await queryBuilder.getCount();
+    const { entities } = await queryBuilder.getRawAndEntities();
+    const students: StudentsResponse[] = [];
+    Logger.log(entities);
+    entities.forEach(entity => {
+      students.push({
+        studentId: entity.id,
+        createdAt: entity.createdAt,
+        details: {
+          grades: {
+            courseCompletion: entity.grades.courseCompletion,
+            courseEngagement: entity.grades.courseEngagement,
+            projectDegree: entity.grades.projectDegree,
+            teamProjectDegree: entity.grades.teamProjectDegree,
+          },
+          employmentExpectations: {
+            expectedTypeWork: entity.profile.expectedTypeWork,
+            targetWorkCity: entity.profile.targetWorkCity,
+            expectedContractType: entity.profile.expectedContractType,
+            expectedSalary: entity.profile.expectedSalary,
+            canTakeApprenticeship: entity.profile.canTakeApprenticeship,
+            monthsOfCommercialExp: entity.profile.monthsOfCommercialExp,
+          },
+        },
+      });
+    });
+
+    const pageMetaDto = new PageMetaDto({ itemCount, pageOptionsDto });
+
+    return new PageDto(students, pageMetaDto);
+  }
+
+  private getAllStudentsQuery(pageOptionsDto: PageOptionsDto): SelectQueryBuilder<Student> {
```

Comment:
czy uda CI się pobrać tylko tych studentów którzy mają utworzony profil? Bo u mnie w bazie testowałem i wywala 500, bo nie u wszystkich mam utworzony profil. A faktycznie tak będize. Że póki Student się nie zaloguje i nie uzupełni danych to nie będzie wyświetlać się jego profil rekruterom